### PR TITLE
Scrape Swiss Results

### PR DESCRIPTION
**Pull request by ochsnerd**
Adds a scraper for the results of the swiss federation (https://swiss-olympic-weightlifting.ch/resultats/)

`check_db.py` is happy with the generated files.

I'm having trouble getting the docker composers to run to test the result further.

Building the backend-docker fails unless I copy `event_data` into `backend`. After that the backend runs and I get a reasonable resonse on `localhost:8080/leaderboard`

Building the frontend-docker fails unless I bump the node-version of the base-image. If I do that, the frontend starts but there's a type-error in fetchLifterData.ts

### CODE REVIEW

**Comment by euanwm** [comment]
Hey @ochsnerd I'll take a look at this today. Apologies for the docker issues, it was in my backlog to update the process for local dev testing.

**Comment by euanwm** [comment]
@ochsnerd I've taken a look at how you're scraping the data and it would probably be easier to query directly from the backend that they're using to build the table on the frontend.

If you take a look at the network transactions when the page is building itself, you'll see a big POST request with some form-data hit an endpoint.

I copied this data into a Postman request and got the data directly in JSON format. 

image

If you want to refactor your current solution to reduce the amount of code you have then I'm happy to wait.

I can deal with the backend & frontend changes but if run the below commands and let me know if you still hit issues with local development then I can spend some time fixing it.

```bash
make build_backend
make build_frontend
./backend/backend &
cd frontend
npm run dev
```

**Comment by euanwm** [comment]
I've created a new issue for the project as we're missing out the gender categories for some lifters and some lifters such as 		Ronny Demaldi (YOB 1989) is technically a Masters lifter and should be put in the relevant Masters category.

Does Switzerland have a Masters category or is that handled by a separate federation?

**Comment by ochsnerd** [comment]
Thanks for having a look.

That you found the endpoint is very cool, I'll gladly refactor the code to use it directly.

I'll let you know when I get around to it, next weekend is probably realistic.

**Comment by euanwm** [comment]
@ochsnerd Not a problem. I am travelling this week so my time on this will be limited so next weekend works well for me to help make any supporting changes.

**Comment by ochsnerd** [comment]
The only 'special' competitions I'm aware of in Switzerland are for Junior lifters, AFAIK the Masters athletes just compete in the open comps (we're a small country).

I can look into mapping the athletes into their appropriate categories myself based on birth year and comp date.

By the way, do I need to 'normalize' the category names into the ones found in `gender_categories.json`? Because at the moment I just leave the strings as they are...

**Comment by euanwm** [comment]
@ochsnerd Yes, please normalize the categories into the ones from `gender_categories.json` but ignore the "Open" categories as they're only used in the US database in lieu of "Senior".

I believe anyone born in 1990 will now be a Masters (35 - 39) lifter.

**Comment by ochsnerd** [comment]
I got around to looking at the code again. Sadly I think getting the data directly from their backend will not work, because each request contains a nonce that is somehow generated in the browser and becomes invalid after some time (https://stackoverflow.com/questions/78067381/scraping-a-website-with-dynamic-wdtnonce-parameter). This is a minimal script to make the request if you're interested:

```python
import requests
import json

url = "https://swiss-olympic-weightlifting.ch/wp-admin/admin-ajax.php?action=get_wdtable&table_id=5"

data = {
    "draw": "1",
    "start": "0",
    "length": "-1",
    "wdtNonce": "dc68fe7675",
}

response = requests.post(url, data=data)

if response.status_code == 200:
    try:
        json_data = response.json()
        print(len(json_data["data"]))
        print(json_data["data"][0])
        for k, v in json_data.items():
            if k == "data":
                print(k[:5])
            print(v)
    except json.JSONDecodeError:
        print("Failed to decode JSON response")
else:
    print(f"Request failed with status code {response.status_code}")
    print(response.text)
```

**Comment by ochsnerd** [comment]
About Categories: I have to admit that I just kind of gave up, for now everyone's just a senior. The 'categories' that are part of the swiss data are kind of a pain to work with. Some entries are missing their category completely, and this screenshot are categories from the same competition:

image

We'll try to talk to the Swiss Federation to get them to expose their data a bit more cleanly

**Comment by euanwm** [comment]
Awesome contribution, thanks!
